src/backend/distributed/metadata/metadata_sync.c

```diff
@@ -395,24 +550,91 @@ NodeDeleteCommand(uint32 nodeId)
+static char *
+LocalGroupIdUpdateCommand(uint32 groupId)
 {
-	char *ownerResetCommand = NULL;
-	char *metadataCommand = NULL;
-	Oid relationId = cacheEntry->relationId;
+	StringInfo updateCommand = makeStringInfo();
 
-	List *commandList = GetTableDDLEvents(relationId);
+	appendStringInfo(updateCommand, "UPDATE pg_dist_local_group SET groupid = %d",
+					 groupId);
 
-	ownerResetCommand = TableOwnerResetCommand(relationId);
-	commandList = lappend(commandList, ownerResetCommand);
+	return updateCommand->data;
+}
 
-	metadataCommand = DistributionCreateCommand(cacheEntry);
-	commandList = lappend(commandList, metadataCommand);
 
-	return commandList;
+/*
+ * MarkNodeHasMetadata function sets the hasmetadata column of the specified worker in
+ * pg_dist_node to true.
+ */
+static void
+MarkNodeHasMetadata(char *nodeName, int32 nodePort, bool hasMetadata)
+{
+	const bool indexOK = false;
+	const int scanKeyCount = 2;
+
+	Relation pgDistNode = NULL;
+	TupleDesc tupleDescriptor = NULL;
+	ScanKeyData scanKey[scanKeyCount];
+	SysScanDesc scanDescriptor = NULL;
+	HeapTuple heapTuple = NULL;
+	Datum values[Natts_pg_dist_node];
+	bool isnull[Natts_pg_dist_node];
+	bool replace[Natts_pg_dist_node];
+
+	pgDistNode = heap_open(DistNodeRelationId(), RowExclusiveLock);
+	tupleDescriptor = RelationGetDescr(pgDistNode);
+
+	ScanKeyInit(&scanKey[0], Anum_pg_dist_node_nodename,
+				BTEqualStrategyNumber, F_TEXTEQ, CStringGetTextDatum(nodeName));
+	ScanKeyInit(&scanKey[1], Anum_pg_dist_node_nodeport,
+				BTEqualStrategyNumber, F_INT8EQ, Int32GetDatum(nodePort));
+
+	scanDescriptor = systable_beginscan(pgDistNode, InvalidOid, indexOK,
+										NULL, scanKeyCount, scanKey);
+
+	heapTuple = systable_getnext(scanDescriptor);
+	if (!HeapTupleIsValid(heapTuple))
+	{
+		ereport(ERROR, (errmsg("could not find valid entry for node \"%s:%d\"",
+							   nodeName, nodePort)));
+	}
+
+	memset(replace, 0, sizeof(replace));
+
+	values[Anum_pg_dist_node_hasmetadata - 1] = BoolGetDatum(hasMetadata);
+	isnull[Anum_pg_dist_node_hasmetadata - 1] = false;
+	replace[Anum_pg_dist_node_hasmetadata - 1] = true;
+
+	heapTuple = heap_modify_tuple(heapTuple, tupleDescriptor, values, isnull, replace);
+	simple_heap_update(pgDistNode, &heapTuple->t_self, heapTuple);
+
+	CatalogUpdateIndexes(pgDistNode, heapTuple);
+
+	CitusInvalidateRelcacheByRelid(DistNodeRelationId());
+
+	CommandCounterIncrement();
+
+	systable_endscan(scanDescriptor);
+	heap_close(pgDistNode, NoLock);
+}
+
+
+/*
+ * TruncateTriggerCreateCommand creates a SQL query calling worker_create_truncate_trigger
+ * function, which creates the truncate trigger on the worker.
+ */
+static char *
+TruncateTriggerCreateCommand(Oid relationId)
+{
+	StringInfo triggerCreateCommand = makeStringInfo();
+	char *tableName = generate_qualified_relation_name(relationId);
+
+	appendStringInfo(triggerCreateCommand,
+					 "SELECT worker_create_truncate_trigger('%s')",
+					 tableName);
```

Comment:
Oops, I meant quote_literal_cstr. These are two different quotes. Also, the template should not contain quotes:

```
"SELECT worker_create_truncate_trigger(%s)", quote_literal_cstr(tableName))"
```